src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -1345,6 +1445,11 @@ private void AssignPatternVariables(BoundPattern pattern)
         public override BoundNode VisitBlock(BoundBlock node)
         {
             DeclareVariables(node.Locals);
+
+            // First assemble possibly-unassigned reads for all
+            // local functions
+            RecordReadsInLocalFunctions(node);
```

Comment:
For efficiency, I think that's probably correct. For correctness, I don't see how it could make a difference, though.